public/app/plugins/datasource/prometheus/components/VariableQueryEditor.tsx

```diff
@@ -164,15 +171,21 @@ export const PromVariableQueryEditor = ({ onChange, query, datasource }: Props)
     }
   };
 
+  const onLabelNamesMatchChange = (regex: string) => {
```

Comment:
Did we want to only call onchange with the blur event?

Reply:
I would argue so, running a query on change only works on prometheus instances where you don't mind running extra or partial queries. Onblur gives you time to write whatever regex/metric names you want without pinging the server before you're ready.